experimental/dds/tree2/src/feature-libraries/default-field-kinds/optionalField.ts

```diff
@@ -105,42 +218,50 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
 		invertChild: NodeChangeInverter,
 		reviver: NodeReviver,
 	): OptionalChangeset => {
-		const inverse: OptionalChangeset = {};
+		// Changes to the child that existed in this field before `change` was applied.
+		let originalChildChanges: NodeChangeset | undefined;
+		const inverseChildChanges = new ChildChangeMap<NodeChangeset>();
+		if (change.childChanges !== undefined) {
+			for (const [id, childChange] of change.childChanges) {
+				if (id === "self" && change.fieldChange !== undefined) {
+					originalChildChanges = invertChild(childChange, 0);
+				} else {
+					inverseChildChanges.set(
+						// This makes assumptions about how sandwich rebasing works
+						id,
+						invertChild(childChange, 0),
+					);
+				}
+			}
+		}
+
+		const selfChanges = change.fieldChange?.newContent?.changes;
+		if (selfChanges !== undefined) {
+			inverseChildChanges.set("self", invertChild(selfChanges, 0));
+		}
+
+		const inverse: OptionalChangeset = {
+			childChanges:
+				inverseChildChanges.size > 0
+					? Array.from(inverseChildChanges.entries())
+					: undefined,
```

Comment:
This may leave the `childChanges` field populated with `undefined`. We generally try to avoid populating the field in that case. It gives a cannonical representation that is terser.

Reply:
Alex brought up the fact that it would make our production code simpler if we always set optiontional field (meaning we'd set them to undefined where we currently avoid setting them).
This would make our test cases more wordy but it does sound worth trying out. So feel free to ignore this.